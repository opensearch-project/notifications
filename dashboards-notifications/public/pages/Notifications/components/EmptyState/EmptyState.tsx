/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiButton, EuiEmptyPrompt, EuiLink, EuiText } from '@elastic/eui';
import React from 'react';
import { DOCUMENTATION_LINK, ROUTES } from '../../../../utils/constants';

interface EmptyStateProps {
  // false if user has no channels, true if no matching search results
  channels: boolean;
}

export function EmptyState(props: EmptyStateProps) {
  return (
    <>
      {props.channels ? (
        <EuiEmptyPrompt
          title={<h2>No notifications insights to display</h2>}
          body={
            <EuiText>
              Adjust your filter or create a channel.{' '}
              <EuiLink href={DOCUMENTATION_LINK} target="_blank" external>
                Learn more
              </EuiLink>
            </EuiText>
          }
        />
      ) : (
        <EuiEmptyPrompt
          title={<h2>You have no channels set up</h2>}
          body={
            <EuiText>
              <div>
                Create a channel, and then configure actions to get
                notifications from features like Alerting and Index Management.{' '}
                <EuiLink href={DOCUMENTATION_LINK} target="_blank" external>
                  Learn more
                </EuiLink>
              </div>
            </EuiText>
          }
          actions={
            <EuiButton href={`#${ROUTES.CREATE_CHANNEL}`} fill>
              Create channel
            </EuiButton>
          }
        />
      )}
    </>
  );
}
