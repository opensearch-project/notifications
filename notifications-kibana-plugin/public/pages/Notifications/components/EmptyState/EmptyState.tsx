/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiButton, EuiEmptyPrompt, EuiLink, EuiText } from '@elastic/eui';
import React from 'react';
import { DOCUMENTATION_LINK, ROUTES } from '../../../../utils/constants';

interface EmptyStateProps {
  // false if user has no channels, true if no matching search results
  channels: boolean;
}

export function EmptyState(props: EmptyStateProps) {
  return (
    <>
      {props.channels ? (
        <EuiEmptyPrompt
          title={<h2>No notifications insights to display</h2>}
          body={
            <EuiText>
              Adjust your filter settings or create a channel and then configure
              notify actions from supported Kibana features.{' '}
              <EuiLink href={DOCUMENTATION_LINK} target="_blank" external>
                Learn more
              </EuiLink>
            </EuiText>
          }
        />
      ) : (
        <EuiEmptyPrompt
          title={<h2>You have no channels set up</h2>}
          body={
            <EuiText>
              <div>
                Create a channel and then configure notify actions from
                supported Kibana features such as Alerting and Index Management.
              </div>
              <br />
              <div>
                Dashboard will generate insights on the notifications sent to
                all of your channels.
              </div>
              <br />
              <div>
                Read about{' '}
                <EuiLink href={DOCUMENTATION_LINK} target="_blank" external>
                  Get started with Notifications
                </EuiLink>
              </div>
            </EuiText>
          }
          actions={
            <EuiButton href={`#${ROUTES.CREATE_CHANNEL}`} size="s">
              Create channel
            </EuiButton>
          }
        />
      )}
    </>
  );
}
