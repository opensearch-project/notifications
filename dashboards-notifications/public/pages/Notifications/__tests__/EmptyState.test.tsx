/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { fireEvent, render } from '@testing-library/react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { EmptyState } from '../components/EmptyState/EmptyState';

describe('<EmptyState /> spec', () => {
  configure({ adapter: new Adapter() });

  it('renders the component with channels configured', () => {
    const utils = render(
      <EmptyState channels />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('renders the component without channels configured', () => {
    const utils = render(
      <EmptyState channels={false} />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

});
