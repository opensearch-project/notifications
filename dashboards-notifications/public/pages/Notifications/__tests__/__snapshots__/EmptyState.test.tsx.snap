// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> spec renders the component with channels configured 1`] = `
<div
  class="euiEmptyPrompt"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <h2
      class="euiTitle euiTitle--medium"
    >
      No notifications insights to display
    </h2>
    <div
      class="euiSpacer euiSpacer--m"
    />
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        Adjust your filter or create a channel.
         
        <a
          class="euiLink euiLink--primary"
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-hidden="true"
            aria-label="External link"
            class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </a>
      </div>
    </div>
  </span>
</div>
`;

exports[`<EmptyState /> spec renders the component without channels configured 1`] = `
<div
  class="euiEmptyPrompt"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <h2
      class="euiTitle euiTitle--medium"
    >
      You have no channels set up
    </h2>
    <div
      class="euiSpacer euiSpacer--m"
    />
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <div>
          Create a channel, and then configure actions to get notifications from features like Alerting and Index Management.
           
          <a
            class="euiLink euiLink--primary"
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn more
            <svg
              aria-label="External link"
              class="euiIcon euiIcon--small euiLink__externalIcon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13 8.5a.5.5 0 111 0V12a2 2 0 01-2 2H4a2 2 0 01-2-2V4a2 2 0 012-2h3.5a.5.5 0 010 1H4a1 1 0 00-1 1v8a1 1 0 001 1h8a1 1 0 001-1V8.5zm-5.12.339a.5.5 0 11-.706-.707L13.305 2H10.5a.5.5 0 110-1H14a1 1 0 011 1v3.5a.5.5 0 11-1 0V2.72L7.88 8.838z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <a
    class="euiButton euiButton--primary euiButton--fill"
    href="#/create-channel"
    rel="noreferrer"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Create channel
      </span>
    </span>
  </a>
</div>
`;
